### ISSUE TITLE
Blocklist slowing down index creation

### ISSUE THREAD

**Issue by poornimanayar** [umbraco/Umbraco-CMS#15056]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.7.0

### Bug summary

We have a site with over 8k nodes. The site relies heavily on the blocklist. Out of these 8k nodes, 6k nodes are of products, the information for which comes in from a PIM and is stored on the node. The data that comes in is in JSON format, so each key-value pair is stored as a block list item. I am using labels as the property editor, but a typical block list item looks as shown below with the property containing upto 30-40 of them.
![image](https://github.com/umbraco/Umbraco-CMS/assets/3941753/7844ee4f-3668-4e56-875c-9849b5f59470)

We have noticed that a index rebuild can take upto an hour. its not just external index but internal index also seems to be affected by this issue.

This was raised as a Cloud Support issue when we noticed index taking too long to build. But the support team, after further investigation, has told us that this might need CMS optimisation and therefore needs an issue created. 

This however has put a big blocker in our plans though for the website :-(

### Specifics

I am raising this issue as we have been talking to @dhedgepeth and Jeyhun regarding this 

### Steps to reproduce

As mentioned above

---
_This item has been added to our backlog AB#34225_

**Comment by dawoe** [comment]
Hi @poornimanayar 

We are noticing performance issues as well on a site that uses the block list editor a lot. See https://github.com/umbraco/Umbraco-CMS/issues/14936

We started noticing a huge performance decrease after we upgraded to 10.4.0 which contains the new block grid. I think some logic for getting values for the block list have been refactored to prevent duplication between Block List and Block Grid.

Dave

**Comment by poornimanayar** [comment]
Thats very good comment @dawoe It didnt strike my eye when I was researching what is going on as my issue was with the indexes. However I did mention to the cloud support that I can see my block list items taking a while to load in the backoffice. Maybe related.

**Comment by poornimanayar** [comment]
Thanks to @mistyn8 who pointed me to this issue, the code that @kjac mentions here has sped up the index generation https://github.com/umbraco/Umbraco-CMS/issues/14096#issuecomment-1508425584

**Comment by poornimanayar** [comment]
I also excluded the block list fields from index during `TransformingIndexValues` event and that seems to have helped a lot atleast in the dev env on CLoud. Yet to test it on live.

**Comment by miguelcrpinto** [comment]
@poornimanayar how did you measure the index creation? You just measured the time it took to rebuild via the backoffice UI?

We have a V8.18.12 instance with way more nodes (I'd say more than 15k) and that often takes **minutes** to publish pages with lots of blocks in a block list property. But in our case rebuilding the indexes is quite fast when triggered from the backoffice (at least the "Rebuild index" button shows up again quite fast after triggering the re-index)

**Comment by poornimanayar** [comment]
Yes, In our case it was just not rebuilding and if I went back in about 45mins later it would be there. But having tested the fix, it seems to be working for us. The index creation has certainly gained speed

**Comment by kjac** [comment]
Closing this one as having been resolved by #14287 then 👍
